utils/splunk/splunk_telemetry_base.py

```diff
@@ -33,12 +57,29 @@ def check(self, instance):
         if not instance.initial_time_done(current_time):
             self.log.debug("Skipping splunk metric/event instance %s, waiting for initial time to expire" % url)
             return
+        initial_token_flag = True
 
         self.load_status()
         instance.update_status(current_time, self.status)
 
         try:
-            self._auth_session(instance)
+            if authentication and 'token_auth' in authentication:
+                self.log.debug("Using token based authentication mechanism")
+                persist_token_key = instance.instance_config.base_url + "token"
+                token = self.status.data.get(persist_token_key)
+                if token is None:
+                    # Since this is first time run, pick the token from conf.yaml
+                    token = authentication["token_auth"].get('initial_token')
+                if self.is_token_expired(instance, token):
+                    msg = "Current in use authentication token is expired. Please provide a valid token in the YAML " \
+                          "and restart the Agent"
+                    raise TokenExpiredException(msg)
+                if self.need_renewal(instance, token, initial_token_flag):
```

Comment:
Is `initial_token_flag` ever False?

Reply:
Might a test be useful? Like testing the number of invocation counts on requesting new tokens? We don't want to generate new tokens when we shouldn't. Also, to prevent that in the future when we make changes. It looks now that we always generate a new token since `initial_token_flag` is always True and because of that will cause `need_renewal` to return True.